### PR TITLE
improve inline_preedit and color_scheme

### PR DESCRIPTION
**Pull request by tuberry**
Let `inline_preedit` be what it should be (aka. `单行模式` in Weasel ), at the same time, do not apply any `color_scheme` when the user has not set it. 

image

Related to #82 and #83

### CODE REVIEW

File: rime_engine.c [hunk 1]
```diff
@@ -360,14 +362,14 @@ static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
     }
   }
 
-  if (inline_text) {
+  if (inline_preedit) {
     ibus_engine_update_preedit_text(
-        (IBusEngine *)rime_engine, inline_text, inline_cursor_pos, TRUE);
```
Comment (lotem) [inline rime_engine.c:365]: It's nice to have this option but pernsonally I don't see it as an improvement over the current behaviour (more WYSIWYG to me).
I wish we can make it more complicated by supporting both options:
https://github.com/rime/weasel/blob/b60b5b128487489036ca8ce8b9ce9442aa5daca6/include/WeaselCommon.h#L135
https://github.com/rime/weasel/blob/ee9128b973fbbb48d874a28bbc1e90c3913f28f7/output/data/weasel.yaml#L19

File: rime_settings.c [hunk 2]
```diff
@@ -6,11 +6,11 @@
 extern RimeApi *rime_api;
 
 static struct ColorSchemeDefinition preset_color_schemes[] = {
+  { NULL, 0, 0 },
   { "aqua", 0xffffff, 0x0a3dfa },
   { "azure", 0xffffff, 0x0a3dea },
   { "ink", 0xffffff, 0x000000 },
-  { "luna", 0x000000, 0xffff7f },
-  { NULL, 0, 0 }
```
Comment (lotem) [inline rime_settings.c:13]: The NULL `color_scheme_id` marks the end of the array. This change does not work.

File: rime_engine.c [hunk 3]
```diff
@@ -308,6 +308,7 @@ static void ibus_rime_engine_update(IBusRimeEngine *rime_engine)
   gboolean inline_preedit =
       g_ibus_rime_settings.embed_preedit_text && context.commit_text_preview;
   gboolean highlighting =
+      (g_ibus_rime_settings.color_scheme->color_scheme_id != NULL) &&
```
Comment (lotem) [inline rime_engine.c:311]: you change made this always NULL and disabled the preedit color scheme feature entirely. I guess the following code path for `highlighting == true` case is not tested.

File: rime_settings.c (hunk 2 above)
Comment (tuberry) [reply rime_settings.c:13]: Sorry, I didn't read the codes carefully.

File: rime_engine.c (hunk 3 above)
Comment (tuberry) [reply rime_engine.c:311]: Yes, I haven't tested that.

File: rime_engine.c (hunk 1 above)
Comment (lotem) [reply rime_engine.c:365]: I see. PREVIEW2 is better.

Another issue you might not have run into is the partially selected state.
That's when you select "你" for "nihao", then the original "preview" style will apply the hightlight to the remainder "好", and the auxiiliary text is also updated to just include the not converted part "hao".
To take that into consideration with PREVIEW2, it is best to left-align the remainer pinyin (axiliary text: "hao" in the above case) to the end of partially selected text ("你"), if that's possible.

To summarize, `inline_curosr_pos` can be set to `highlighting_start` in the partially selected case.

Comment (tuberry) [reply rime_engine.c:365]: Got it. 
image
Another question: in COMPOSITION, the highlight will change when the cursor is moving to left or right. Should I hightlight the whole preedit text with the color_scheme or use the reverse color_cheme to distinguish the two parts?
COMPOSITION1:
image
COMPOSITION2:
image

**Comment by lotem** [comment]
The new code still have bugs, like in sentence style input (朙月拼音·語句流).
I find it a little bit hard to reason about the code since some variables has changed meanings than in the old workflow. I ended up rewriting code for updating display content in `rime_engine.c`.
Then added config option `style/cursor_pos` to control if the inline cursor is at the beginning or at the logical insertion point.
